providers/dns/ibmcloud/internal/wrapper.go

```diff
@@ -66,7 +67,27 @@ func getDomainID(service services.Dns_Domain, domain string) (*int, error) {
 		return r.Id, nil
 	}
 
-	return nil, fmt.Errorf("no data found of domain: %s", domain)
+	// The domain was not found by name. For subdomains this is not unusal in softlayer.
+	// So in case a subdomain like sub.toplevel.domain was used try again using the
+	// parent domain (strip the first part in the domain string -> toplevel.domain).
+	domainParts := strings.Split(domain, ".")
+	if len(domainParts) > 2 {
+		return getDomainID(service, getParentDomain(domainParts))
+	} else {
+		return nil, fmt.Errorf("no data found for domain: %s", domain)
+	}
```

Comment:
```suggestion
	parent, err := getParentDomain(domain)
	if err != nil {
		return nil, err
	}

	return getDomainID(service, parent)
```

Reply:
My understanding is that Cut will return the before and after parts in regards to the sperator.
So if domain string is "toplevel.domain" before will be "toplevel" and after will be "domain" (found will be true).

So for this case the getDomainID function is called allthough it does not make sense to call it because "domain" itself will not yield a result. This is why the initial code had the len(domainParts) > 2 check.

This will probably not cause harm but a call to getDomainID that will always fail.

Thanks for looking into this.

Reply:
FYI my algo has the same behavior as yours, but it is 10 times faster.